File: Lista/quest08/agregacao.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def exibir_empregados(self):
        print(f"Empregados da {self.nome}:")
        for emp in self.empregados:
            print(f"Nome: {emp.nome}, Cargo: {emp.cargo}, Salário: {emp.salario:.2f}")

if __name__ == "__main__":
    empresa = Empresa("Empresa X")

    empregado1 = Empregado("Alice", "Desenvolvedora", 5000)
    empregado2 = Empregado("Bob", "Designer", 4000)

    empresa.adicionar_empregado(empregado1)
    empresa.adicionar_empregado(empregado2)

    empresa.exibir_empregados()

File: Lista/quest08/agregacao.go
package main

import "fmt"

type Empregado struct {
	Nome  string
	Cargo string
	Salario float64
}

type Empresa struct {
	Nome      string
	Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(emp Empregado) {
	e.Empregados = append(e.Empregados, emp)
}

func (e Empresa) ExibirEmpregados() {
	fmt.Printf("Empregados da %s:\n", e.Nome)
	for _, emp := range e.Empregados {
		fmt.Printf("Nome: %s, Cargo: %s, Salário: %.2f\n", emp.Nome, emp.Cargo, emp.Salario)
	}
}

func main() {
	empresa := Empresa{Nome: "Empresa X"}

	empregado1 := Empregado{Nome: "Alice", Cargo: "Desenvolvedora", Salario: 5000}
	empregado2 := Empregado{Nome: "Bob", Cargo: "Designer", Salario: 4000}

	empresa.AdicionarEmpregado(empregado1)
	empresa.AdicionarEmpregado(empregado2)

	empresa.ExibirEmpregados()
}
